client/src/routes/Session/hooks/useMuteAudio.ts

```diff
@@ -0,0 +1,32 @@
+import {useCallback, useContext} from 'react';
+import {useTranslation} from 'react-i18next';
+import {ExerciseSlide} from '../../../../../shared/src/types/Content';
+import {MicrophoneOffIcon} from '../../../lib/components/Icons';
+
+import {DailyContext} from '../../../lib/daily/DailyProvider';
+import useSessionNotificationsState from '../state/sessionNotificationsState';
+
+const useMuteAudio = () => {
+  const {t} = useTranslation('Screen.Session');
+  const {muteAll} = useContext(DailyContext);
+  const addNotification = useSessionNotificationsState(
+    state => state.addNotification,
+  );
+
+  const conditionallyMuteParticipants = useCallback(
+    (playing: boolean, currentSlide: ExerciseSlide | undefined) => {
+      if (playing && currentSlide?.type !== 'sharing') {
+        muteAll();
+        addNotification({
+          text: t('notifications.muted'),
+          Icon: MicrophoneOffIcon,
+        });
```

Comment:
I believe this notification will only be shown for the host that mutes. Maybe we need to move this logic into `SessionNotifications` instead?

Reply:
In the old `useMuteAudioListener` it only adds notification if you're the session host, so I guess that never was shown to the other participants

Reply:
Oh then it was a bug. We can fix that separately. Can you write a todo?

Reply:
Or actually it wasn’t a bug since it was the clients themselves that muted because of the state change. So it would turn up on all clients.

Reply:
Tested it now without my change, only the host got the notification

Reply:
Even if the clients muted themselves only the condition for the notification was if you where the session host